qbft/spectest/tests/controller_spectest.go

```diff
@@ -180,3 +189,52 @@ func (test *ControllerSpecTest) runInstanceWithData(
 
 	return lastErr
 }
+
+func (test *ControllerSpecTest) overrideStateComparison(t *testing.T) {
+	basedir, _ := os.Getwd()
+	path := filepath.Join(basedir, "generate", "state_comparison", reflect.TypeOf(test).String(), fmt.Sprintf("%s.json", test.TestName()))
+	byteValue, err := os.ReadFile(path)
+	require.NoError(t, err)
+
+	sc := make([]*qbft.Controller, len(test.RunInstanceData))
+	require.NoError(t, json.Unmarshal(byteValue, &sc))
+
+	for i, runData := range test.RunInstanceData {
+		runData.ControllerPostState = sc[i]
+
+		r, err := sc[i].GetRoot()
+		require.NoError(t, err)
+		runData.ControllerPostRoot = hex.EncodeToString(r[:])
+	}
+}
+
+func (test *ControllerSpecTest) GetPostState() (interface{}, error) {
```

Comment:
if we change something in the run function we need to change it here as well..

Reply:
yes